### ISSUE TITLE
Life Support resources being consumed for empty Vessel

### ISSUE THREAD

**Issue by JPLRepo** [BobPalmer/USI-LS#200]
USI LS does not seem to be handling vessels that become devoid of Kerbals (it's happened before https://github.com/BobPalmer/USI-LS/pull/87).

So with my DeepFreeze mod, which as far as I can tell is doing the right thing by calling  LifeSupportManager.UntrackKerbal when it freezes a kerbal. The kerbal is removed from USI LS tracking in the UI and seems ok... but...
Single kerbal vessel. Put into space. It starts with 4500 supplies.
![2017-01-09 11_33_37-kerbal space program](https://cloud.githubusercontent.com/assets/9981382/21755023/08fcafcc-d660-11e6-9c15-953f274d68a8.png)

I freeze the kerbal. He is removed from USI LS tracking successfully.
![2017-01-09 11_33_50-kerbal space program](https://cloud.githubusercontent.com/assets/9981382/21755024/16e2ccf2-d660-11e6-8fcc-7eba9ad41626.png)

I timewarp forward at a rapid rate. and I witness the Life Support supplies going down... with no kerbal on-board to consume them
![2017-01-09 11_34_08-kerbal space program](https://cloud.githubusercontent.com/assets/9981382/21755028/2a2d3efa-d660-11e6-9069-91e0775bd5c3.png)

Then when I thaw the kerbal, for a brief second the LS supplies go down the amount he would have consumed if he was in the vessel awake.
![2017-01-09 11_34_21-kerbal space program](https://cloud.githubusercontent.com/assets/9981382/21755031/425b0a0c-d660-11e6-9c92-bd39bb2d8187.png)

and then suddenly they go back again... but not to the 4500 supplies that it should be.
![2017-01-09 11_34_29-kerbal space program](https://cloud.githubusercontent.com/assets/9981382/21755032/4bec0724-d660-11e6-9572-4e97cb46a4df.png)

End result. Removing all kerbals from the vessel and untracking them results in USI LS supplies still being consumed.

**Comment by JPLRepo** [comment]
Further info.
When the last Kerbal in Untracked in USI LS, USI does not update the VesselSupplyInfo.NumCrew. this is because a VesselChange GameEvent occurs which triggers ModuleLifeSuportSystem.SetVesselDirty which in turn triggers UpdateVesselInfo which sets _currentCrew to the Vessel.GetCrewCount() cache. IE: 0.

This in turn means that the if(_currentCrew == 0) return; occurs in ModuleLifeSupportSystem.FixedUpdate and the VesselSupplyInfo.NumCrew is never updated... in fact the VesselSupplyInfo is never updated until a crew member enters the vessel again. So critical here is the fact that the time since resources were last consumed, etc are not reset.....
![2017-01-09 16_53_53-kerbal space program](https://cloud.githubusercontent.com/assets/9981382/21758227/a411ed60-d68d-11e6-9e09-05d12c28608d.png)
 So when a kerbal re-enters the vessel. It starts consuming resources from the last time there these values were updated.
There seems to be a check for this attempted in the LifeSupportMonitor class but this only happens with the GUI is OPEN, and even then it doesn't trip up and remove the vessel tracking or reset it because it checks the VesselSupplyInfo.NumCrew == 0, which it never is, it never goes below VesselSupplyInfo.NumCrew = 1.
This is very easy to fix from a DeepFreeze perspective.
I can just call LifeSupportManager. UntrackVessel if the vessel is in fact empty, which I will implement.

However, I suspect look at this that you will have similar issues going on with kerbals transferring and decoupling a vessel and coming back to the first vessel potentially. I didn't keep going to test that or check how the code handles those use cases.

**Comment by JPLRepo** [comment]
Actually it is not easy for DeepFreeze to fix. 
1) The suggestion to re-track the Kerbal manually does not work.
2) Attempting to Untrack the Vessel also does not work, this is because USI LS immediately starts tracking the vessel again. Because the NumCrew is not 0, and also because Habitat processing??? seems to re-track vessels in the area.

The only way to fix this is to actually correct the bug in ModuleLifeSupportsystem.FixedUpdate()
There is no way NumCrew can be set to zero because of this:
if (_currentCrew == 0) return;
occurs before further processing that actually does update this field.

I also noticed in further process it does another check:
if (_currentCrew > 0)
                {
does a bunch of processing....
then there is an Else statement?????
else
                {
                    VesselStatus.LastECCheck = Planetarium.GetUniversalTime();
                    VesselStatus.LastFeeding = Planetarium.GetUniversalTime();
                    VesselStatus.LastUpdate = Planetarium.GetUniversalTime();
                }

Except this else would never execute because of the earlier Guard?? check of zero and return.
So I changed that earlier If statement to:-

if (_currentCrew == 0)
            {
                VesselStatus.VesselName = vessel.vesselName;
                VesselStatus.NumCrew = vessel.GetCrewCount();
                VesselStatus.CrewCap = vessel.GetCrewCapacity();
                VesselStatus.LastECCheck = Planetarium.GetUniversalTime();
                VesselStatus.LastFeeding = Planetarium.GetUniversalTime();
                VesselStatus.LastUpdate = Planetarium.GetUniversalTime();
                LifeSupportManager.Instance.TrackVessel(VesselStatus);
                LastUpdateTime = Planetarium.GetUniversalTime();
                return;
            }

Which is slightly more than the dead Else statement.
Because 
a:- we actually want to update the NumCrew field to Zero when there are in fact zero crew on board.
b:- Even when I only moved the dead else statements Kerbals woken up in a vessel or added to a vessel suddenly started using all the resources again.
This is because the VesselModule LastUpdatetime is not being updated even when there are no crew on board. So immediately that there are it starts consuming resources since this time.
